src/h5web/dataset-visualizer/DatasetVisualizer.tsx

```diff
@@ -1,29 +1,34 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import { HDF5Entity, HDF5Collection } from '../providers/models';
 import styles from './DatasetVisualizer.module.css';
 import { useValue } from '../providers/hooks';
-import { isBaseType, isSimpleShape } from '../providers/type-guards';
-import MatrixVis from './MatrixVis';
+import VisualizationSelector from './VisualizationSelector';
+import { Vis } from './models';
+import { getSupportedVis } from './utils';
+import Visualization from './Visualization';
 
 interface Props {
   dataset: HDF5Entity<HDF5Collection.Datasets>;
 }
 
 function DatasetVisualizer(props: Props): JSX.Element {
   const { dataset } = props;
-  const { id, type, shape } = dataset;
 
-  const value = useValue(id);
+  const value = useValue(dataset.id);
+
+  const supportedVis = useMemo(() => getSupportedVis(dataset), [dataset]);
```

Comment:
Do you fear that `getSupportedVis` will become expensive to call ?

Reply:
Not hugely, but it could. Since it's up to the visualizations to say whether they're supported, `getSupportedVis` is a bit like a black box, so better to play it safe. It also means we don't have to worry if we do decide that we need a very complex check (e.g. verifying the type of every value in an nD array for instance).